src/sql/workbench/contrib/dashboard/browser/widgets/insights/actions.ts

```diff
@@ -23,6 +22,14 @@ export class RunInsightQueryAction extends Action {
 	}
 
 	public run(context: InsightActionContext): Promise<boolean> {
-		return this.instantiationService.invokeFunction(openNewQuery, context.profile, undefined, RunQueryOnConnectionMode.executeQuery).then(() => true, () => false);
+		let queryString: string = undefined;
+		if (context.insight && context.insight.query) {
+			if (typeof (context.insight.query) === 'string') {
```

Comment:
should just use the typing helper functions

Reply:
where are the typing helper functions?

Reply:
https://github.com/microsoft/azuredatastudio/blob/master/src/vs/base/common/types.ts#L32